src/groupByKey.ts

```diff
@@ -2,6 +2,19 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(
+  items: T[],
+  key: keyof T,
+): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    groupedItems[`${item[key]}`] = [];
+  });
+
+  items.forEach((item: T) => {
+    groupedItems[`${item[key]}`].push(item);
+  });
```

Comment:
Let's make our logic simpler, why to iterate array twice if we can do it only once 🙂